src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -210,6 +223,67 @@ def run(
                 )
                 raise unexpected_error from e
 
+    def _copy_previous_run_result(
+        self,
+        resume_from_run_storage: AbstractBatchRunStorage,
+        resume_from_run_output_dir: Path,
+        batch_inputs: List,
+        output_dir: Path,
+    ) -> List[LineResult]:
+        """Duplicate the previous debug_info from resume_from_run_storage and output from resume_from_run_output_dir
+        to the storage of new run,
+        return the list of previous line results for the usage of aggregation and summarization.
+        """
+        previous_run_results = []
+        previous_run_output = self._load_outputs(resume_from_run_output_dir) if resume_from_run_output_dir else []
+        previous_run_output_dict = {
+            each_line_output[LINE_NUMBER_KEY]: each_line_output for each_line_output in previous_run_output
+        }
+
+        if resume_from_run_output_dir:
+            for filename in os.listdir(resume_from_run_output_dir):
```

Comment:
There's no guarantee that all the referenced image would be directly under the root folder. E.g. there could be (in the future) line in the json line file like {"image": "./some_child_folder/1.png"}
So please do recursive copy.

Please also put the logic in a separate util function.
It could be something like this (from GPT):
```
import os
import shutil

def copy_files_except(src_dir, dst_dir, exclude_file):
    """
    Copy all files from src_dir to dst_dir recursively, excluding a specific file
    directly under the root of src_dir.
    
    Args:
    - src_dir (str): Source directory path.
    - dst_dir (str): Destination directory path.
    - exclude_file (str): Name of the file to exclude from copying.
    """
    # Ensure the destination directory exists
    os.makedirs(dst_dir, exist_ok=True)
    
    # Walk through the source directory
    for root, dirs, files in os.walk(src_dir):
        # Calculate the relative path to the current directory
        rel_path = os.path.relpath(root, src_dir)
        current_dst_dir = os.path.join(dst_dir, rel_path)
        
        # Ensure the current destination directory exists
        os.makedirs(current_dst_dir, exist_ok=True)
        
        # Copy each file in the current directory, except the excluded file
        for file in files:
            if rel_path == '.' and file == exclude_file:
                continue  # Skip the excluded file
            src_file_path = os.path.join(root, file)
            dst_file_path = os.path.join(current_dst_dir, file)
            shutil.copy2(src_file_path, dst_file_path)
```

Reply:
Extracted the code to _copy_file_except and move it to promptflow._utils.utils.